### PR TITLE
update Status.java with encode/decode.

### PR DESCRIPTION
**Pull request by gurunrao**
update Status.java with encode/decode[ encode/decode are same as at https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java ]
Signed-off-by: gurunandan.rao@oracle.com <gurunandan.rao@oracle.com>

### CODE REVIEW

**Comment by jtulach** [comment]
What is your motivation for the change? Can you accompany it with an (initially failing) test?

**Comment by gurunrao** [comment]
> What is your motivation for the change? Can you accompany it with an (initially failing) test?

Tests are failing with `java.lang.NoSuchMethodError: com.sun.javatest.Status.decode(Ljava/lang/String;)Ljava/lang/String;`, when we update to latest version https://github.com/jtulach/netbeans-apitest/releases/tag/release-1.4

Details of JAF TCK run can be found here - https://ci.eclipse.org/jakartaee-tck/job/guru/job/jaf-tck-guru/job/sigsjar/3/console

**Comment by gurunrao** [comment]
@jtulach - Please review.

**Comment by gurunrao** [comment]
> What is your motivation for the change? Can you accompany it with an (initially failing) test?

@jtulach - currently we are replicating changes from https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java for encode/decode, we don't have unit-test code for the same.

**Comment by JaroslavTulach** [comment]
I see [a failure](https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java):
```
[javatest.batch] Beginning Test:  javasoft/sqe/tests/api/jakarta/activation/ActivationDataFlavor/equals_Test.java#equals_Test
[javatest.batch] Exception in thread "DefaultTestRunner:Worker-0:0_StreamCopier_3" java.lang.NoSuchMethodError: com.sun.javatest.Status.decode(Ljava/lang/String;)Ljava/lang/String;
[javatest.batch] 	at com.sun.javatest.lib.ProcessCommand$StreamCopier.run(ProcessCommand.java:444)
```
How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?

**Comment by gurunrao** [comment]
> I see [a failure](https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java):
> 
> ```
> [javatest.batch] Beginning Test:  javasoft/sqe/tests/api/jakarta/activation/ActivationDataFlavor/equals_Test.java#equals_Test
> [javatest.batch] Exception in thread "DefaultTestRunner:Worker-0:0_StreamCopier_3" java.lang.NoSuchMethodError: com.sun.javatest.Status.decode(Ljava/lang/String;)Ljava/lang/String;
> [javatest.batch] 	at com.sun.javatest.lib.ProcessCommand$StreamCopier.run(ProcessCommand.java:444)
> ```
> 
> How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?
Follow these steps for reproducing error with CLI:
```
git clone https://github.com/gurunrao/jaf-tck
cd jaf-tck
git checkout sig-error-demo
export WORKSPACE=$PWD
./docker/build_activationtck.sh
./docker/run_activationtck.sh`
```

**Comment by gurunrao** [comment]
> I see [a failure](https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java):
> 
> ```
> [javatest.batch] Beginning Test:  javasoft/sqe/tests/api/jakarta/activation/ActivationDataFlavor/equals_Test.java#equals_Test
> [javatest.batch] Exception in thread "DefaultTestRunner:Worker-0:0_StreamCopier_3" java.lang.NoSuchMethodError: com.sun.javatest.Status.decode(Ljava/lang/String;)Ljava/lang/String;
> [javatest.batch] 	at com.sun.javatest.lib.ProcessCommand$StreamCopier.run(ProcessCommand.java:444)
> ```
> 
> How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?

@jtulach - For reproducing above errors, Please following CLI steps:

```
git clone https://github.com/gurunrao/jaf-tck
cd jaf-tck
git pull
git checkout update-sigtest-error
export WORKSPACE=$PWD
./docker/build_activationtck.sh
./docker/run_activationtck.sh
```

**Comment by gurunrao** [comment]
> I see [a failure](https://github.com/openjdk/jtharness/blob/master/src/com/sun/javatest/Status.java):
> 
> ```
> [javatest.batch] Beginning Test:  javasoft/sqe/tests/api/jakarta/activation/ActivationDataFlavor/equals_Test.java#equals_Test
> [javatest.batch] Exception in thread "DefaultTestRunner:Worker-0:0_StreamCopier_3" java.lang.NoSuchMethodError: com.sun.javatest.Status.decode(Ljava/lang/String;)Ljava/lang/String;
> [javatest.batch] 	at com.sun.javatest.lib.ProcessCommand$StreamCopier.run(ProcessCommand.java:444)
> ```
> 
> How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?

@JaroslavTulach - have updates this PR with unit test for Status.java encode/decode functions.

File: src/test/java/com/sun/javatest/StatusTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.sun.javatest;
```
Comment (jtulach) [inline src/test/java/com/sun/javatest/StatusTest.java:26]: Missing license. Probably copy it from the [APITest](https://github.com/jtulach/netbeans-apitest/blob/master/src/test/java/org/netbeans/apitest/APITest.java) file.

**Comment by jtulach** [comment]
> > How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?
> 
> @jtulach - For reproducing above errors, Please following CLI steps:

OK. Thank you...

> 
> ```
> git clone https://github.com/gurunrao/jaf-tck
> cd jaf-tck
> git pull
> git checkout update-sigtest-error
> export WORKSPACE=$PWD
> ./docker/build_activationtck.sh

This creates `activation-tck-2.1.0.zip` which contains `activation-tck/sigtest.jar` - same file as [sigtest-maven-plugin-1.4.jar](https://repo1.maven.org/maven2/org/netbeans/tools/sigtest-maven-plugin/1.4/sigtest-maven-plugin-1.4.jar).

> ```
> ./docker/run_activationtck.sh
> ```

I'd like to see the single test failure. I am able to run:
```
jaf-tck/activation-tck$ java -cp classes/:javatest.jar:../jakarta.activation-api.jar  javasoft.sqe.tests.api.jakarta.activation.ActivationDataFlavor.ActivationDataFlavor_Test
testForActivationDF1: Passed. ActivationDataFlavor(String,String) test succeeded
testForActivationDF2: Passed. ActivationDataFlavor(Class,String) test succeeded
testForActivationDF3: Passed. ActivationDataFlavor(Class,String,String) test succeeded
STATUS:Passed.test cases: 3; all passed
```
but the test passes. I am not going to dig deeper today.

**Comment by gurunrao** [comment]
> > > How does the test look like? What does it do? Can I reproduce that on CLI (or in unit tests) somehow?
> > 
> > 
> > @jtulach - For reproducing above errors, Please following CLI steps:
> 
> OK. Thank you...
> 
> > ```
> > git clone https://github.com/gurunrao/jaf-tck
> > cd jaf-tck
> > git pull
> > git checkout update-sigtest-error
> > export WORKSPACE=$PWD
> > ./docker/build_activationtck.sh
> > ```
> 
> This creates `activation-tck-2.1.0.zip` which contains `activation-tck/sigtest.jar` - same file as [sigtest-maven-plugin-1.4.jar](https://repo1.maven.org/maven2/org/netbeans/tools/sigtest-maven-plugin/1.4/sigtest-maven-plugin-1.4.jar).
> 
> > ```
> > ./docker/run_activationtck.sh
> > ```
> 
> I'd like to see the single test failure. I am able to run:
> 
> ```
> jaf-tck/activation-tck$ java -cp classes/:javatest.jar:../jakarta.activation-api.jar  javasoft.sqe.tests.api.jakarta.activation.ActivationDataFlavor.ActivationDataFlavor_Test
> testForActivationDF1: Passed. ActivationDataFlavor(String,String) test succeeded
> testForActivationDF2: Passed. ActivationDataFlavor(Class,String) test succeeded
> testForActivationDF3: Passed. ActivationDataFlavor(Class,String,String) test succeeded
> STATUS:Passed.test cases: 3; all passed
> ```
> 
> but the test passes. I am not going to dig deeper today.

@JaroslavTulach - failures are not reproducible from command line. The failure are reproducible with javatest batch testsuite, executed using commands mentioned here https://github.com/jtulach/netbeans-apitest/pull/4#issuecomment-966163590 . Also javatest.jar has complete `com/sun/javatest/Status.java` file, but sigtest.jar produced from repository doesn't have complete `com/sun/javatest/Status.java`.When both 'javatest.jar' 'sigtest.jar' are added to classpath  and run with javatest batch, results are failures. 

javatest.jar should have Status.java[ as it does currently] and not sigtest.jar, hence have made changes [https://github.com/jtulach/netbeans-apitest/compare/master...gurunrao:sig-test-remove-javatest-code?expand=1] for removal of javatest code[ Status.java, Test.java] and add javatest dependency to the pom.xml, which resolves the issue.

**Comment by JaroslavTulach** [comment]
> failures are not reproducible from command line

Aren't? Really? Or we just don't know the `java` command parameters that we should pass in? I'd like to see the classpath and JVM options that are used when the test fails.

**Comment by gurunrao** [comment]
> > failures are not reproducible from command line
> 
> Aren't? Really? Or we just don't know the `java` command parameters that we should pass in? I'd like to see the classpath and JVM options that are used when the test fails.

There is no point in reproducing the issue, since as I said before Status.java code shouldn't be part of the sigtest.jar, its part of javatest.jar. Please let me know if I can raise a PR for removing Status.java,Test.java?

**Comment by jtulach** [comment]
Removing `Status` & `Test` as proposed in #5 sounds way better!

**Pull request closed without merging**